bootstraptest/test_yjit.rb

```diff
@@ -1421,6 +1421,81 @@ def make_str(foo)
   uplus_str.frozen?
 }
 
+# String-subclass objects should behave as expected inside string-interpolation via concatstrings
+assert_equal 'monkeys, yo!', %q{
+  class MyString < String
+    # This is a terrible idea in production code, but we'd like YJIT to match CRuby
+    def to_s
+      super + ", yo!"
+    end
+  end
+
+  m = MyString.new('monkeys')
+  "#{m.to_s}"
+
+  raise "String-subclass to_s should not be called for interpolation" if "#{m}" != 'monkeys'
+  raise "String-subclass to_s should be called explicitly during interpolation" if "#{m.to_s}" != 'monkeys, yo!'
+
+  m.to_s
+}
+
+# String-subclass objects should behave as expected for string equality
+assert_equal 'a', %q{
+  class MyString < String
+    # This is a terrible idea in production code, but we'd like YJIT to match CRuby
+    def ==(b)
+      "#{self}_" == b
+    end
+  end
+
+  ma = MyString.new("a")
+
+  raise "Not dispatching to string-subclass equality!" if ma == "a" || ma != "a_"
+  raise "Incorrectly dispatching for String equality!" if "a_" == ma || "a" != ma
+  raise "Error in equality between string subclasses!" if ma != MyString.new("a_")
+  # opt_equality has an explicit "string always equals itself" test, but should never be used when == is redefined
+  raise "Error in reflexive equality!" if ma == ma
```

Comment:
Generally, we haven't been using raise in any of these tests. We also try to keep them short, because if YJIT side-exits, it won't be testing what we need.

Reply:
Okay. I found some other raises in the file, but I could probably find a way to remove these. This would be a lot more lines if I need to re-declare the assert block and string class for every comparison, though.

Reply:
Alright. Will accept these as-is :)